spec/policies/comment_policy_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CommentPolicy do
+  subject { described_class.new(user, comment) }
+
+  let(:user) { nil }
+  let(:comment_user) { create(:user) }
+  let(:task_user) { create(:user) }
+  let(:access_level) { :private }
```

Comment:
I am not sure which line you mean but I think all of them are used. Am I missing something? The :user is used in the subject definition, the :comment_user in the comment and the :task_user and :access_level are used in the :task which is needed for the comment. We could inline the :task_user in the creation of :task since we dont change :task_user in the specs.

Reply:
Indeed, all those variables are used throughout the test file. 

Actually, I haven't noticed the `task_user` thing, where we could, indeed, inline the assignment. But I wanted to double-check with the `access_level` defined in line 11 (formerly) / 10 (as of yesterday). Let me explain shortly:

The test file has two main contexts, depending on the fact whether a user can access or cannot access the task. In the first line of each context, you define an `access_level`. Hence, in those two contexts, the `access_level` is given through these "inner" definitions. Besides these two contexts, there is no other test. Hence, the "outer" `access_level` is never used to create a task (but only the two "inner" ones). As a consequence, we can omit the "outer" `access_level`. Due to the nature of RSpec, the order of `let` definitions is not important, allowing to use a name in "outer" context, even though it is only defined in (all) "inner" contexts.